### PR TITLE
Create `SubmitRecover` to handle panic recovery in workerpool tasks if desired

### PR DESCRIPTION
**Pull request by cameronbrill**
allows user to overwrite the panic handler function with `workerpool.SetPanicHandler(func())`

### CODE REVIEW

File: workerpool.go [hunk 1]
```diff
@@ -123,6 +131,24 @@ func (p *WorkerPool) SubmitWait(task func()) {
 	<-doneChan
 }
 
+// SubmitRecover enqueues the given function and defers a panic recover func.
```
Comment (cameronbrill) [inline workerpool.go:134]:
```suggestion
// SubmitRecover enqueues the given function and defers a call to `p.panicHandler`.
```

**Comment by cameronbrill** [comment]
@gammazero is this something someone can review? if you don't think it's fitting for your project, I'll keep it in my fork.

File: LICENSE [hunk 2]
```diff
@@ -1,6 +1,9 @@
 The MIT License (MIT)
 
-Copyright (c) 2016 Andrew J. Gillis
+Copyright for portions of project gammazero/workerpool are held by
+Andrew J. Gillis, 2016 as part of project gammazero/workerpool. All
+other copyright for project highlight-run/workerpool are held by Highlight
```
Comment (gammazero) [inline LICENSE:5]: This is specific to highlight-run so would not be suitable for merge into this repo.

File: README.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-[![Go Report Card](https://goreportcard.com/badge/github.com/gammazero/workerpool)](https://goreportcard.com/report/github.com/gammazero/workerpool)
-[![codecov](https://codecov.io/gh/gammazero/workerpool/branch/master/graph/badge.svg)](https://codecov.io/gh/gammazero/workerpool)
-[![License](https://img.shields.io/badge/License-MIT-blue.svg)](https://github.com/gammazero/workerpool/blob/master/LICENSE)
+[![GoDoc](https://pkg.go.dev/badge/github.com/highlight-run/workerpool)](https://pkg.go.dev/github.com/highlight-run/workerpool)
+[![Build Status](https://github.com/highlight-run/workerpool/actions/workflows/go.yml/badge.svg)](https://github.com/highlight-run/workerpool/actions/workflows/go.yml)
+[![Go Report Card](https://goreportcard.com/badge/github.com/highlight-run/workerpool)](https://goreportcard.com/report/github.com/highlight-run/workerpool)
+[![codecov](https://codecov.io/gh/highlight-run/workerpool/branch/master/graph/badge.svg)](https://codecov.io/gh/highlight-run/workerpool)
+[![License](https://img.shields.io/badge/License-MIT-blue.svg)](https://github.com/highlight-run/workerpool/blob/master/LICENSE)
```
Comment (gammazero) [inline README.md:7]: These changes are for the highlight-run fork, and not suitable for this PR.

File: README.md [hunk 4]
```diff
@@ -14,18 +14,21 @@ This implementation builds on ideas from the following:
 - http://nesv.github.io/golang/2014/02/25/worker-queues-in-go.html
 
 ## Installation
-To install this package, you need to setup your Go workspace.  The simplest way to install the library is to run:
+
+To install this package, you need to setup your Go workspace. The simplest way to install the library is to run:
+
 ```
-$ go get github.com/gammazero/workerpool
+$ go get github.com/highlight-run/workerpool
```
Comment (gammazero) [inline README.md:21]: This is for the highlight-run fork, and cannot merge this into this repo.

File: pacer/pacer.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ import "time"
 //     })
 //
 //     wp := workerpool.New(5)
-//     wp.Submit(pacedTask)
+//     wp.submit(pacedTask)
```
Comment (gammazero) [inline pacer/pacer.go:34]: This example does not work, since it tries to call a non-exported function.

File: pacer/pacer_test.go [hunk 6]
```diff
@@ -5,7 +5,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/gammazero/workerpool"
+	"github.com/highlight-run/workerpool"
```
Comment (gammazero) [inline pacer/pacer_test.go:8]: Cannot merge this.

File: workerpool.go [hunk 7]
```diff
@@ -112,7 +112,7 @@ func (p *WorkerPool) Stopped() bool {
 // period until there are no more idle workers.  Since the time to start new
 // goroutines is not significant, there is no need to retain idle workers
 // indefinitely.
-func (p *WorkerPool) Submit(task func()) {
+func (p *WorkerPool) submit(task func()) {
```
Comment (gammazero) [inline workerpool.go:115]: No longer exporting `Submit` will break existing code that uses Workerpool.

File: go.mod [hunk 8]
```diff
@@ -1,4 +1,4 @@
-module github.com/gammazero/workerpool
+module github.com/highlight-run/workerpool
```
Comment (gammazero) [inline go.mod:1]: This is certainly not a change that could be merged.

**Review by gammazero — COMMENTED** [review]
Most of the changes here are specific to the `highlight-run` fork and are not suitable for merging into this repo.

Also, I would prefer not to have a `SubmitRecover` function in place of `Submit`:
- The caller can provide this in their own function passed to `Submit`
- If the caller does not supply the function, then can provider their own recover wrapper to wrap the function that is passed to Submit.
- `Submit` must still be available since some applications my not want recovery, or are submitting a function that already handles recovery. 

I think it is best to close this PR and leave these changes in your fork.

**Comment by gammazero** [comment]
Closing, since I think it is more appropriate for this change to live in the fork.  Thank you for your interest in this project.

**Pull request closed without merging**